src/main/java/frc/robot/RobotConstants.java

```diff
@@ -8,6 +14,52 @@
         public static final int BALL_SENSOR = 0;
     }
 
+    public interface MOTOR_SETTINGS {
+        public static TalonSettings INTAKE = TalonSettingsBuilder.defaults()
+                .withCurrentLimits(20, 15, 200)
+                .brakeInNeutral()
+                .withDirections(false, false)
+                .noMotorOutputLimits()
+                .noMotorStartupRamping()
+                .useSpeedControl()
+                .build();
+        
+        public static TalonSettings ELEVATOR = TalonSettingsBuilder.defaults()
+                .withCurrentLimits(20, 15, 200)
+                .brakeInNeutral()
+                .withDirections(false, false)
+                .noMotorOutputLimits()
+                .noMotorStartupRamping()
+                .useSpeedControl()
+                .build();
+        
+        public static TalonSettings HOOD = TalonSettingsBuilder.defaults()
+                .withCurrentLimits(1, 1, 1)
+                .brakeInNeutral()
+                .withDirections(false, false)
+                .noMotorOutputLimits()
+                .noMotorStartupRamping()
+                .usePositionControl()
+                .withGains(4, 2.56 * 3, 0, 0)
+                .withMotionProfile(1000, 1000, 5)
+                .enableLimitSwitch(false).build();
+        
+        public static SparkMaxSettings SHOOTER = SparkMaxSettingsBuilder.defaults()
+                .withCurrentLimits(30)
+                .coastInNeutral()
+                .withDirections(false, false)
+                .limitMotorOutputs(1, -1)
+                .withMotorRampUpOnStart(0.5)
+                .useSmartMotionControl()
+                .withPositionGains(0.000015, 12, 4e-4, 0.0)
+                .useAccelerationStrategy(CANPIDController.AccelStrategy.kSCurve)
+                .withMaxVelocity(6000)
+                .withMaxAcceleration(3000)
+                .withClosedLoopError(15)
+                .build();
+     
```

Comment:
I like this a lot better. It is much cleaner and readable code.

Reply:
yeah i agree. if it works i think we should move all the motor settings into constants like this. i just did a couple